### DISCUSSION TITLE
Problem with Windows Installer for Captive Runtime Bundled Package

### DISCUSSION THREAD

**Discussion by XiaoGen-Taiwan** [airsdk/Adobe-Runtime-Support#2169] [General]
Hi, I'm trying to generate a Windows Installer with InnoSetup for my captive runtime bundled package.
But when I open the Installer an Info Dialog opens with the message: "This application requires a version of Adobe AIR which cannot be found ... ".
Anyone, an idea?

Packaging the app as an AIR app or a native installer works.

**Comment by 2jfw** [comment]
I think the error will e.g. occur when the value of `<content>` under `<initialWindow>` in `App Descriptor` points to an unresolvable/invalid file. Try to check this

**Reply by ajwfrost** [reply]
That's worth checking; I've seen that message before mostly when there's a set-up that doesn't include the AIR captive runtime though. So if you have a native installer application but the AIR app depends on anything beyond 32, then the installer goes looking for an upgrade package and doesn't find one.. so then it gives that error message.

Does the installer definitely include the Adobe AIR subfolder/DLL? and how are you packaging the whole thing up -> I would assume you use ADT with the "bundle" target to generate the folder that you then want to distribute, and are using "InnoSetup" to deploy that folder in an installer form?

**Reply by XiaoGen-Taiwan** [reply]
> I think the error will e.g. occur when the value of `<content>` under `<initialWindow>` in `App Descriptor` points to an unresolvable/invalid file. Try to check this

Thanks @2jfw and @ajwfrost  for your replies. Na, the `<content>` section points to the correct swf file.

@ajwfrost, Sorry, I wasn't clear enough in my question ... the captive runtime bundled package did work as well. 
My problem was actually how to setup InnoSetup in the right way. Because of receiving that Error message about 'AIR can not be found' I was thinking there are additional steps necessary for creating a captive runtime bundled installer.
Instead I made a mistake (figured it actually out by your comment about the subfolder/DLL).

// The bundled captive runtime package
MyApplication (bundle folder)
| Adobe AIR
| META-INF 
MyApplication.exe
MyApplication.swf
mimetype

My mistake was that I was pointing to the executable and added all other files and folders manually within the InnoSetup wizard.

The right way is:
Pointing to the executable but instead of adding all other files and folders,
the bundle folder itself has to be added.
